python/gherkin3/token_matcher.py

```diff
@@ -50,13 +91,11 @@ def match_StepLine(self, token):
             title = token.line.get_rest_trimmed(len(keyword))
             self._set_token_matched(token, 'StepLine', title, keyword)
             return True
-
         return False
 
     def match_Comment(self, token):
         if not token.line.startswith('#'):
             return False
-
```

Comment:
In theory the change from using `[]` as default argument in https://github.com/Zearin/gherkin3/commit/176cae0e23f6071a994a477a15d1972b35d1f9ff could fix a problem. In practice I do not think it can happen, since `token.matched_items` is never modified, only iterated over. Still, to avoid warning bells ringing when people reads that code, it could be wise to use the default value `None` instead. When I look at that method now I would change the assignment to: `token.matched_items = items or []` in order to make `token.matched_items` possible to iterate over at all times - instead of using an if-statement that conditionally assigns `[]` to `items`. That keeps `_set_token_matched` focused on assigning the token attributes.

Using `str.join` instead of string concatenation with `+` as in https://github.com/Zearin/gherkin3/commit/0b1b7c55c85fbf73e463fecb6b0a590c15128e76 is more efficient (https://waymoot.org/home/python_string/). On the other hand since `TokenFormatterBuilder` is only used in the acceptance tests, maybe efficiency is not the most important thing there.